### PR TITLE
Enable content projection in kirby-list

### LINKED ISSUE #1813: [Enhancement] Enable declarative list building via projected content
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the enhancement
To provide better control over list, I would like to be able to have a list version where I do not supply data to the list via the `items` input, but instead via content projection, where I declare the list something like this: 

```
<kirby-list>
    <kirby-item>
        ...
    </kirby-item>

    <kirby-item>
        ...
    </kirby-item>

    <kirby-item>
        ...
    </kirby-item>
</kirby-list>
```
This requirement originates from experiments with virtual scroll described in #1794, where it would suit the use-case better if consuming projects could have more fine grained control over their lists, so they can add necessary html and css to e.g. the items, to supply the correct information of item heights and similar to the virtual scroll algorithms.

Along these lines, this approach will give most likely make for a more flexible list, that can support many more use-cases, without having to account for these directly in the kirby source code, as teams themselves can make small adjustments.

This approach is also in line with the standard HTML approach, [ionic](https://ionicframework.com/docs/api/list), and how [Material Design](https://material.io/components/lists/web#single-line-list) and [Carbon](https://material.io/components/lists/web#single-line-list) does it.

## Describe the solution you'd like
This solution should be strictly limited to adding an option for content projection to the list, as #1814  and #1815 will add proper styling, and #1816 will handle sections. This could be as simple as using the existing template if the items input has been set, and using a new `<ng-content></ng-content>` template if items is not set. 

## Have you considered any alternatives?
We could consider to keep the current structure, and try to implement use cases like virtual scrolling ourselves, continuing the work on #1606. 

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Refinement

- [ ] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#ux-refinement); do not proceed until this is done.
- [ ] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [master branch](https://github.com/kirbydesign/designsystem/tree/master) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

### PR DESCRIPTION
**Pull request by RasmusKjeldgaard**
## Which issue does this PR close?
This PR closes #1813 

## What is the new behavior?
In addition to supplying an array of items, items can now be specified declaratively in HTML, between the `<kirby-list>` tags. 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Are there any additional context?

This is a precursor to enable the work on #1883.

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [ ] Request that the changes are code-reviewed 
- [ ] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be automatically merged to master via [automerge](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/automatically-merging-a-pull-request).

### CODE REVIEW

**Comment by RasmusKjeldgaard** [comment]
There is no documentation in this PR. But I guess we should keep it that way until we fully support at least the simple list with correct look and feel?

**Comment by RasmusKjeldgaard** [comment]
Moved showDivider-stuff to [poc/1813-show-divider-declarative-list](https://github.com/kirbydesign/designsystem/tree/poc/1813-show-divider-declarative-list), as we need to discuss how to apply such conditional styling to items in the new list structure.

File: libs/designsystem/src/lib/components/list/list.component.spec.ts [hunk 1]
```diff
@@ -49,7 +54,7 @@ const TEST_ITEMS: any[] = [
   },
 ];
 
-describe('ListComponent', () => {
+describe('ListComponent with array of items', () => {
```
Comment (MadsBuchmann) [inline libs/designsystem/src/lib/components/list/list.component.spec.ts:57]:
```suggestion
describe("ListComponent using 'item' input property", () => {
```

File: libs/designsystem/src/lib/components/list/list.component.spec.ts [hunk 2]
```diff
@@ -307,3 +312,32 @@ describe('ListComponent', () => {
...
+  });
+
+  beforeEach(() => {
+    spectator = createHost<ListComponent>(`
+      <kirby-list>
+        <kirby-item><h3>Test Item</h3></kirby-item>
+        <kirby-item><h3>Test Item</h3></kirby-item>
+        <kirby-item><h3>Test Item</h3></kirby-item>
+        <kirby-item><h3>Test Item</h3></kirby-item>
+      </kirby-list>`);
```
Comment (MadsBuchmann) [inline libs/designsystem/src/lib/components/list/list.component.spec.ts:331]:
```suggestion
      <kirby-list>
        <kirby-item>Test Item</kirby-item>
        <kirby-item>Test Item</kirby-item>
        <kirby-item>Test Item</kirby-item>
        <kirby-item>Test Item</kirby-item>
      </kirby-list>`);
```

Just to keep it simple(r)

File: libs/designsystem/src/lib/components/list/list-experimental/list-experimental.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+describe('ListExperimental', () => {
+  let spectator: SpectatorHost<ListExperimentalComponent>;
+
+  const createHost = createHostFactory({
+    component: ListExperimentalComponent,
+    declarations: [
+      ListExperimentalComponent,
+      MockComponent(IonList),
+      MockComponent(IonItem),
+      ItemComponent,
```
Comment (MadsBuchmann) [inline libs/designsystem/src/lib/components/list/list-experimental/list-experimental.component.spec.ts:18]: We're not mocking ItemComponent. Is this actually an integration test?

File: libs/designsystem/src/lib/components/list/list-experimental/list-experimental.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+  it('should create', () => {
+    expect(spectator.component).toBeTruthy();
+  });
+
+  it('should render items in list', () => {
+    const items = spectator.queryHostAll('kirby-item');
+
+    expect(items).toHaveLength(4);
+  });
```
Comment (MadsBuchmann) [inline libs/designsystem/src/lib/components/list/list-experimental/list-experimental.component.spec.ts:40]: We're not mocking items. Is this actually an integration test?

File: libs/designsystem/src/lib/components/list/list-experimental/list-experimental.component.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'kirby-list-experimental',
+  templateUrl: './list-experimental.component.html',
```
Comment (MadsBuchmann) [inline libs/designsystem/src/lib/components/list/list-experimental/list-experimental.component.ts:5]: I like that we're not including any css files as of now. Truly separating structure from styling.

File: libs/designsystem/src/lib/components/list/list-experimental/list-experimental.component.spec.ts (hunk 4 above)
Comment (RasmusKjeldgaard) [reply libs/designsystem/src/lib/components/list/list-experimental/list-experimental.component.spec.ts:40]: Good catch, let us mock the items!

File: libs/designsystem/src/lib/components/list/list-experimental/list-experimental.component.ts (hunk 5 above)
Comment (RasmusKjeldgaard) [reply libs/designsystem/src/lib/components/list/list-experimental/list-experimental.component.ts:5]: Wohooo 💯

**Review by MadsBuchmann — DISMISSED** [review]
Me using the new declarative list:

**Comment by RasmusKjeldgaard** [comment]
Hey

**Review by MadsBuchmann — APPROVED** [review]

**Pull request merged**